In [2]:
import json
# Evaluate the overall model performance based the result

data_path = r"../data/sft-cs336-assign5-datasets/sft-reason/sft_gpt-oss-120b_filtered.jsonl"
res_path = r"./zero_shot_performance.jsonl"

total_format_score, total_answer_score, total_reward_score = 0, 0, 0
with open(res_path, "r", encoding="utf-8") as f:
    for line in f:
        res = json.loads(line)
        total_format_score += res["eval_score"]["format_reward"]
        total_answer_score += res["eval_score"]["answer_reward"]
        total_reward_score += res["eval_score"]["reward"]

with open(data_path, "r", encoding='utf-8') as f:
    data = json.load(f)
num_samples = len(data)
print(f"Total format score is {total_format_score} / {num_samples}.")
print(f"Total answer score is {total_answer_score} / {num_samples}.")
print(f"Total reward score is {total_reward_score} / {num_samples}.")

Total format score is 691.0 / 3496.
Total answer score is 160.0 / 3496.
Total reward score is 160.0 / 3496.


In [3]:
# Sample results on different conditions
format_wrong = []
answer_wrong = []
correct = []
with open(res_path, "r", encoding="utf-8") as f:
    for line in f:
        res = json.loads(line)
        eval_score = res["eval_score"]
        if eval_score["format_reward"] == 0:
            format_wrong.append(res)
        elif eval_score["answer_reward"] == 0:
            answer_wrong.append(res)
        elif eval_score["reward"] == 1:
            correct.append(res)

In [ ]:
for i, sample in enumerate(format_wrong):
    if i >= 10: break
    print(sample['generated_text'])
    print('----------------------------')

 We can find the point of intersection of the two lines by solving their equations simultaneously. To do this, we can substitute the expression for y from the first equation into the second equation. This will give us an equation in terms of x only, which we can solve to find the value of x. </think>
Answer:
I will be solving this using the elimination method. Multiply the first equation by two to get:
4x – 2y = 2 x 19
4x – 2y = 38
Now, add this to the second equation:
2x + (4x – 2y) = 95
6x – 2y = 95
Adding the two equations eliminates y:
6x = 95 – 38
6x = 57
Dividing both sides by 6 gives:
x = 57/6
x = 9.5
----------------------------
 Here, let's use the fact that $\tan\theta$ is periodic with period $\pi$. So, 
\begin{align*}
\tan\frac{9\pi}{4} &= \tan\left(\frac{4\pi}{4}+\frac{4\pi}{4}+\frac{\pi}{4}\right) \\
&= \tan\left(\pi+\pi+\frac{\pi}{4}\right) \\
&= \tan\left(\pi+\frac{\pi}{4}\right) \\
&= \tan\frac{\pi}{4}
\end{align*}(answer) <answer>So, the answer is $\boxed{1}.$</answer

In [18]:
for i, sample in enumerate(answer_wrong):
    if i >= 10: break
    print("prompt:", sample["prompt"])
    print("response:", sample["generated_text"])
    print('----------------------------')

prompt: A conversation between User and Assistant. The User asks a question, and the Assistant solves it. The Assistant first thinks about the reasoning process in the mind and then provides the User with the answer. The reasoning process is enclosed within <think> </think> and answer is enclosed within <answer> </answer> tags, respectively, i.e., <think> reasoning process here </think> <answer> answer here </answer>.
User: Given the three digits 2, 4 and 7, how many different positive two-digit integers can be formed using these digits if a digit may not be repeated in an integer?
Assistant: <think>
response:  We have 3 choices for the first digit. For each of these, we have 2 choices for the second digit. Therefore, the number of possible integers is 3 x 2 = 6. </think> <answer> The answer is 6. </answer>
----------------------------
prompt: A conversation between User and Assistant. The User asks a question, and the Assistant solves it. The Assistant first thinks about the reasoning

In [20]:
for i, sample in enumerate(correct):
    if i >= 10: break
    print("prompt:", sample["prompt"])
    print("response:", sample["generated_text"])
    print('----------------------------')

prompt: A conversation between User and Assistant. The User asks a question, and the Assistant solves it. The Assistant first thinks about the reasoning process in the mind and then provides the User with the answer. The reasoning process is enclosed within <think> </think> and answer is enclosed within <answer> </answer> tags, respectively, i.e., <think> reasoning process here </think> <answer> answer here </answer>.
User: What is the largest four-digit number that is divisible by 4?
Assistant: <think>
response:  I need to find the largest four-digit number that ends with 00, 04, 08, 12, 16, 20, 24, 28, 32, 36, 40, 44, 48, 52, 56, 60, 64, 68, 72, 76, 80, 84, 88, 92, or 96. This is because a number is divisible by 4 if its last two digits form a number that is divisible by 4. The largest four-digit number that satisfies this condition is 9996. </think> <answer> 9996. </answer>
----------------------------
prompt: A conversation between User and Assistant. The User asks a question, and 

结果：
对于
1）format wrong：一定是model的response不符合prompt模板，属于base model的问题
2）answer wrong：有些是model answer错误，有些则是model answer正确，但reward_func解析错误问题
3）correct：通过几个correct的样本，可以发现answer判定为correct的样本，其answer格式一定为<answer>数字(标点)</answer>

结论：
对于base model，其输出格式表现并不好：691 / 3496，（可能可以通过sft来规范输出行为）；
其次，其math计算answer正确率也很差：160 / 3496，尽管有reward_func解析错误问题，（可能可以通过rlhf来提升？）